inst/@sym/horzcat.m

```diff
@@ -62,7 +62,7 @@
           'return sp.Matrix.hstack(*_proc),'
           };
 
-  varargin = sym(varargin);
+  varargin = cell_array_to_sym(varargin);
```

Comment:
You are getting error not because of these changes. You can get rid of those errors by using "addpath('<path_to_octsympy>/inst/@sym/private')"
.
And yes i am using hg version. I don't think there's a git repo for octave, is there?